lib/models/UserStudyRole.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+const cypher = require('../utils/get-mcrypt-cypher.js')();
+
+function decryptUsername(username) {
+    return cypher.decrypt(new Buffer(username.trim(), 'base64'))
+        .toString().replace(/\0/g, '');
+}
+
+
+module.exports = (bookshelf) => {
+    return bookshelf.extend({
+        tableName: 'casdba.cas_study_user_role_privs',
+        idAttribute: ['username', 'role_id', 'study_id'],
+        role: function() {
+            return this.belongsTo('CasRole', 'role_id');
+        },
+        initialize: function() {
+            this.on('fetched', this.decryptAttributes);
```

Comment:
This overrides the `initialize` callback of the `baseModel`, thus you need to manually perform the same actions that are performed within the baseModel's `initialize` callback. In particular, add this: 
```
this.on('saved', this._utils.saveHist);
```